WordPress/src/main/java/org/wordpress/android/widgets/WPDialogSnackbar.java

```diff
@@ -0,0 +1,157 @@
+package org.wordpress.android.widgets;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.design.widget.Snackbar;
+import android.support.design.widget.Snackbar.SnackbarLayout;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.TextView;
+
+import org.wordpress.android.R;
+
+/**
+ * {@link Snackbar} with {@link android.app.Dialog}-like layout.  The layout views include title, message,
+ * positive button, and negative button.  Any empty or null view is hidden.  The only required view is message.
+ */
+public class WPDialogSnackbar {
+    private Snackbar mSnackbar;
+    private View mContentView;
+
+    private WPDialogSnackbar(@NonNull View view, @NonNull CharSequence text, int duration) {
+        switch (duration) {
+            case Snackbar.LENGTH_INDEFINITE:
+                mSnackbar = Snackbar.make(view, "", Snackbar.LENGTH_INDEFINITE);
+                break;
+            case Snackbar.LENGTH_LONG:
+                mSnackbar = Snackbar.make(view, "", Snackbar.LENGTH_LONG);
+                break;
+            default:
+                mSnackbar = Snackbar.make(view, "", Snackbar.LENGTH_SHORT);
+        }
+
+        // Set underlying snackbar layout.
+        SnackbarLayout snackbarLayout = (SnackbarLayout) mSnackbar.getView();
+        snackbarLayout.setPadding(0, 0, 0, 0);
+
+        // Hide underlying snackbar text and action.
+        TextView snackbarText = snackbarLayout.findViewById(android.support.design.R.id.snackbar_text);
+        snackbarText.setVisibility(View.INVISIBLE);
+        TextView snackbarAction = snackbarLayout.findViewById(android.support.design.R.id.snackbar_action);
+        snackbarAction.setVisibility(View.INVISIBLE);
+
+        mContentView = ((LayoutInflater) view.getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE))
```

Comment:
Android Studio is complaining about a possible NPE for `((LayoutInflater) view.getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE))`, let's do a null check for it using a variable before inflating it.

Reply:
Using `getSystemService(Context.LAYOUT_INFLATER_SERVICE)` does have the chance to return `null`, but `LayoutInflater.from(view.getContext())` does not.  I updated the layout inflater method in 5b2c29b.